### PR TITLE
[AppGW] Adding backend health API.

### PR DESCRIPTION
**Pull request by kagamsft**

### CODE REVIEW

File: src/ResourceManagement/Network/Microsoft.Azure.Management.Network/Generated/ApplicationGatewaysOperations.cs [hunk 1]
```diff
@@ -1346,6 +1346,224 @@ internal ApplicationGatewaysOperations(NetworkManagementClient client)
...
+        /// <param name='expand'>
+        /// expand references resources.
+        /// </param>
+        /// <param name='customHeaders'>
+        /// The headers that will be added to request.
+        /// </param>
+        /// <param name='cancellationToken'>
+        /// The cancellation token.
+        /// </param>
+        public async Task<AzureOperationResponse<ApplicationGatewayBackendHealth>> BackendHealthWithHttpMessagesAsync(string resourceGroupName, string applicationGatewayName, string expand = default(string), Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
```
Comment (shahabhijeet) [inline src/ResourceManagement/Network/Microsoft.Azure.Management.Network/Generated/ApplicationGatewaysOperations.cs]: I am not able to see test coverage for the newly added API's. Please add test coverage.

File: src/ResourceManagement/Network/Network.Tests/Tests/ApplicationGatewayTests.cs [hunk 2]
```diff
@@ -482,6 +482,11 @@ public void ApplicationGatewayApiTest()
                 Assert.Equal(appGwName, getGateway.Name);
                 CompareApplicationGateway(appGw, getGateway);
 
+                //Get AppGw backend health
```
Comment (shahabhijeet) [inline src/ResourceManagement/Network/Network.Tests/Tests/ApplicationGatewayTests.cs]: Please remove commented code from this file.

File: src/ResourceManagement/Network/Microsoft.Azure.Management.Network/Generated/ApplicationGatewaysOperations.cs (hunk 1 above)
Comment (kagamsft) [inline src/ResourceManagement/Network/Microsoft.Azure.Management.Network/Generated/ApplicationGatewaysOperations.cs]: ApplicationGatewayTests.cs has new API covered. Here is the code:

                //Get AppGw backend health
                var backendHealth = networkManagementClient.ApplicationGateways.BackendHealth(resourceGroupName, appGwName);
                Assert.Equal(1, backendHealth.BackendAddressPools.Count);
                Assert.Equal(2, backendHealth.BackendAddressPools[0].BackendHttpSettingsCollection.Count);

File: src/ResourceManagement/Network/Network.Tests/Tests/ApplicationGatewayTests.cs (hunk 2 above)
Comment (kagamsft) [inline src/ResourceManagement/Network/Network.Tests/Tests/ApplicationGatewayTests.cs]: This comment describes the API call. Should be allowed?

**Pull request merged**